Optional dependency tests failing with Aesara

The optional dependency tests are now failing in the Aesara tests but only on pypy and CPython 3.11:
https://github.com/sympy/sympy/runs/7210465165?check_suite_focus=true

The test that fails is this one:
https://github.com/sympy/sympy/blob/997749da1838628eaeb077c7ab32da6666912e13/sympy/printing/tests/test_aesaracode.py#L608-L614

It's only the most recent merge commit on the master branch that shows this failure but it also shows up in all PRs now. Something must have changed in between 14 and 22 hrs ago and it looks like it's the release of Aesara 2.7.5 which was 18 hrs ago:
https://pypi.org/project/aesara/#history

CC @brandonwillard

I think the reason it fails only on pypy and 3.11 is because we don't install pymc there and pymc currently pins `aesara==2.7.4` so it's only the pypy and 3.11 jobs where Aesara 2.7.5 gets installed.

Actually I can reproduce this locally with CPython 3.8 and Aesara 2.7.5:
```python
In [1]: from sympy.printing.aesaracode import aesara_code
WARNING (aesara.tensor.blas): Using NumPy C-API based implementation for BLAS functions.

In [2]: from sympy.printing.tests.test_aesaracode import theq

In [3]:      xt, yt = aesara_code(x, dtypes={x:'complex128'}), aesara_code(y, dtypes={y: 'complex128'
   ...: })
   ...:      from sympy.functions.elementary.complexes import conjugate
   ...:      from aesara.tensor import as_tensor_variable as atv
   ...:      from aesara.tensor import complex as cplx
   ...:      assert theq(aesara_code(y*conjugate(x)), yt*(xt.conj()))
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-3-5104cf8662e7> in <module>
      3 from aesara.tensor import as_tensor_variable as atv
      4 from aesara.tensor import complex as cplx
----> 5 assert theq(aesara_code(y*conjugate(x)), yt*(xt.conj()))

AssertionError: 

In [4]: aesara_code(y*conjugate(x))
Out[4]: Elemwise{mul,no_inplace}.0

In [5]: yt*(xt.conj())
Out[5]: Elemwise{mul,no_inplace}.0

In [6]: theq(aesara_code(y*conjugate(x)), yt*(xt.conj()))
Out[6]: False
```
The `theq` function is supposed to compare two Aesara objects:
https://github.com/sympy/sympy/blob/997749da1838628eaeb077c7ab32da6666912e13/sympy/printing/tests/test_aesaracode.py#L97-L98

--- a/sympy/printing/tests/test_aesaracode.py
+++ b/sympy/printing/tests/test_aesaracode.py
@@ -606,11 +606,12 @@ def test_Relationals():
 
 
 def test_complexfunctions():
-    xt, yt = aesara_code(x, dtypes={x:'complex128'}), aesara_code(y, dtypes={y: 'complex128'})
+    dtypes = {x:'complex128', y:'complex128'}
+    xt, yt = aesara_code(x, dtypes=dtypes), aesara_code(y, dtypes=dtypes)
     from sympy.functions.elementary.complexes import conjugate
     from aesara.tensor import as_tensor_variable as atv
     from aesara.tensor import complex as cplx
-    assert theq(aesara_code(y*conjugate(x)), yt*(xt.conj()))
+    assert theq(aesara_code(y*conjugate(x), dtypes=dtypes), yt*(xt.conj()))
     assert theq(aesara_code((1+2j)*x), xt*(atv(1.0)+atv(2.0)*cplx(0,1)))
 
 

